backends/qualcomm/scripts/download_qnn_sdk.py

```diff
@@ -0,0 +1,247 @@
+            return None
+
+        if QAIRT_URL.endswith(".zip"):
+            print("Extracting ZIP archive...")
+            _extract_zip(archive_path, QAIRT_CONTENT_DIR, SDK_DIR)
+        elif QAIRT_URL.endswith((".tar.gz", ".tgz")):
+            print("Extracting TAR archive...")
+            _extract_tar(archive_path, QAIRT_CONTENT_DIR, SDK_DIR)
+        else:
+            raise ValueError(f"Unsupported archive format: {QAIRT_URL}")
+
+        print(f"Verifying extraction to {SDK_DIR}")
+        if SDK_DIR.exists():
+            print(f"SDK directory exists. Contents:")
+            for item in SDK_DIR.iterdir():
+                print(f"  {item.name}")
+        else:
+            print("ERROR: SDK directory was not created!")
+
+        print(f"Qualcomm SDK extracted to {SDK_DIR}")
+
+    return SDK_DIR
+
+
+def _extract_zip(archive_path, content_dir, target_dir):
+    print(f"Extracting {archive_path} to {target_dir}")
+    print(f"Looking for content in subdirectory: {content_dir}")
+
+    target_dir.mkdir(parents=True, exist_ok=True)
+
+    with zipfile.ZipFile(archive_path, "r") as zip_ref:
+        files_to_extract = [f for f in zip_ref.namelist() if f.startswith(content_dir)]
+
+        for file in tqdm(files_to_extract, desc="Extracting files"):
+            relative_path = pathlib.Path(file).relative_to(content_dir)
+            if relative_path == pathlib.Path("."):
+                continue
+
+            out_path = target_dir / relative_path
+            if file.endswith("/"):
+                out_path.mkdir(parents=True, exist_ok=True)
+            else:
+                out_path.parent.mkdir(parents=True, exist_ok=True)
+                with zip_ref.open(file) as src, open(out_path, "wb") as dst:
+                    shutil.copyfileobj(src, dst)
+
+
+def _extract_tar(archive_path: pathlib.Path, prefix: str, target_dir: pathlib.Path):
+    with tarfile.open(archive_path, "r:gz") as tf:
+        for m in tf.getmembers():
+            if not m.name.startswith(prefix + "/"):
+                continue
+            relpath = pathlib.Path(m.name).relative_to(prefix)
+            if not relpath.parts or relpath.parts[0] == "..":
+                continue
+
+            out_path = target_dir / relpath
+            if m.isdir():
+                out_path.mkdir(parents=True, exist_ok=True)
+            else:
+                out_path.parent.mkdir(parents=True, exist_ok=True)
+                src = tf.extractfile(m)
+                if src is None:
+                    continue
+                with src, open(out_path, "wb") as dst:
+                    dst.write(src.read())
+
+
+LLVM_VERSION = "14.0.0"
+LIBCXX_BASE_NAME = f"clang+llvm-{LLVM_VERSION}-x86_64-linux-gnu-ubuntu-18.04"
+LLVM_URL = f"https://github.com/llvm/llvm-project/releases/download/llvmorg-{LLVM_VERSION}/{LIBCXX_BASE_NAME}.tar.xz"
+REQUIRED_LIBS = [
+    "libc++.so.1.0",
+    "libc++abi.so.1.0",
+    "libunwind.so.1",
+    "libm.so.6",
+    "libpython3.10.so.1.0",
```

Comment:
what is this?

Reply:
It's currently used for internal, can remove if it's not needed in next pr